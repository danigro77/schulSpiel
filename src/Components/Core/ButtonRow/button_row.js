import React, {Component} from 'react'
import {Grid, Cell} from 'react-foundation'

export default class ButtonRow extends Component {

    getCellSize(itemProps) {
        let classes = '';
        if (itemProps.small) {
            classes += `small-${itemProps.small.toString()} `
        }
        if (itemProps.medium) {
            classes += `medium-${itemProps.medium.toString()} `
        }
        if (itemProps.large) {
            classes += `large-${itemProps.large.toString()} `
        }
        return classes;
    }

    renderButtons() {
        return this.props.buttons.map((button, index) => {
            let buttonProps = {
                type: button.type || 'button',
                value: button.text || 'Click me',
                className: 'button',
            };
            if (button.onClick) {
                buttonProps.onClick = button.onClick;
            }
            if (button.klasses) {
                buttonProps.className += ` ${button.klasses}`
            }
            return (
                <Cell className={this.getCellSize(button)} key={index}>
                    <input {...buttonProps} />
                </Cell>
            )
        })
    }

    render() {
        return (
            <Grid className="align-justify">
                {this.renderButtons()}
            </Grid>
        )
    }
}