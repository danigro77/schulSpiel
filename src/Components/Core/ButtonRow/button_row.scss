$base-button-background-color:#1779ba;
$base-button-font-color:#ffffff;

.button {
  color: $base-button-font-color;
  background: $base-button-background-color;
  &.secondaryButton {
    color: $base-button-background-color;
    background: $base-button-font-color;
    border: $base-button-background-color 1px solid;
  }
}

@media screen and (max-width: 500px) {
  .button {
    font-size: 0.8em;
  }
}