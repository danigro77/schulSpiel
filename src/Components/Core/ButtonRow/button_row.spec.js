import React from 'react'
import sinon from 'sinon'

import ButtonRow from './button_row'

const buttons = {
    oneButton: [
        {
            text: 'Spec Test'
        }
    ], multipleButtons: [
        {
            text: 'Spec Test Button 1'
        },
        {
            text: 'Spec Test Button 2',
            type: 'submit',
        }
    ], eventButtons: [
        {
            text: 'Spec Test Event',
            onClick: sinon.spy(),
        },
    ]
};

describe('<BottonRow />', () => {
    describe('snapshots', () => {
        it('render a button row with one button', () => {
            const wrapper = shallow(
                <ButtonRow
                    buttons={buttons.oneButton}
                />
            );
            expect(wrapper).toMatchSnapshot();
        });

        it('render a button row with multiple button', () => {
            const wrapper = shallow(
                <ButtonRow
                    buttons={buttons.multipleButtons}
                    small={6}
                    large={4}
                />
            );
            expect(wrapper).toMatchSnapshot();

        });
    });

    describe('DOM testing', () => {
        it('render a button row with one button', () => {
            const wrapper = mount(
                <ButtonRow
                    buttons={buttons.oneButton}
                />
            );

            expect(wrapper.find('input').length).toEqual(1);
            expect(wrapper.find('input').props().value).toEqual(buttons.oneButton[0].text);
        });

        it('render a button row with multiple button', () => {
            const buttonProps = buttons.multipleButtons;
            const wrapper = mount(
                <ButtonRow
                    buttons={buttonProps}
                />
            );
            expect(wrapper.find('input').length).toEqual(2);
            for (let i=0; i<buttonProps.length; i++) {
                expect(wrapper.find('input').at(i).props().value).toEqual(buttonProps[i].text);
            }
        });

        it('registers a button click when an event is passed in', () => {
            const buttonProps = buttons.eventButtons;
            const wrapper = mount(
                <ButtonRow
                    buttons={buttonProps}
                />
            );
            expect(wrapper.find('input').props().value).toEqual(buttonProps[0].text);
            wrapper.find('input').simulate('click');
            expect(buttonProps[0].onClick.calledOnce).toEqual(true);
        });

    });
});

