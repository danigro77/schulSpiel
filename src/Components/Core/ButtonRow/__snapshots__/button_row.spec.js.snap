// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BottonRow /> snapshots render a button row with multiple button 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonRow
    buttons={
        Array [
            Object {
              "text": "Spec Test Button 1",
            },
            Object {
              "text": "Spec Test Button 2",
              "type": "submit",
            },
          ]
    }
    large={4}
    small={6}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Cell
          className=""
>
          <input
                    className="button"
                    type="button"
                    value="Spec Test Button 1"
          />
</Cell>,
        <Cell
          className=""
>
          <input
                    className="button"
                    type="submit"
                    value="Spec Test Button 2"
          />
</Cell>,
      ],
      "className": "align-justify",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "children": <input
            className="button"
            type="button"
            value="Spec Test Button 1"
/>,
          "className": "",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "button",
            "type": "button",
            "value": "Spec Test Button 1",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "children": <input
            className="button"
            type="submit"
            value="Spec Test Button 2"
/>,
          "className": "",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "button",
            "type": "submit",
            "value": "Spec Test Button 2",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Cell
            className=""
>
            <input
                        className="button"
                        type="button"
                        value="Spec Test Button 1"
            />
</Cell>,
          <Cell
            className=""
>
            <input
                        className="button"
                        type="submit"
                        value="Spec Test Button 2"
            />
</Cell>,
        ],
        "className": "align-justify",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "children": <input
              className="button"
              type="button"
              value="Spec Test Button 1"
/>,
            "className": "",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "button",
              "type": "button",
              "value": "Spec Test Button 1",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "children": <input
              className="button"
              type="submit"
              value="Spec Test Button 2"
/>,
            "className": "",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "button",
              "type": "submit",
              "value": "Spec Test Button 2",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`<BottonRow /> snapshots render a button row with one button 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ButtonRow
    buttons={
        Array [
            Object {
              "text": "Spec Test",
            },
          ]
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <Cell
          className=""
>
          <input
                    className="button"
                    type="button"
                    value="Spec Test"
          />
</Cell>,
      ],
      "className": "align-justify",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "children": <input
            className="button"
            type="button"
            value="Spec Test"
/>,
          "className": "",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "button",
            "type": "button",
            "value": "Spec Test",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Cell
            className=""
>
            <input
                        className="button"
                        type="button"
                        value="Spec Test"
            />
</Cell>,
        ],
        "className": "align-justify",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "children": <input
              className="button"
              type="button"
              value="Spec Test"
/>,
            "className": "",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "button",
              "type": "button",
              "value": "Spec Test",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
